tests/Integration/inc/Engine/Optimization/RUCSS/Admin/Subscriber/cronCleanRows.php

```diff
@@ -0,0 +1,93 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Tests\Integration\inc\Engine\Optimization\RUCSS\Admin\Subscriber;
+
+use WP_Rocket\Tests\Integration\DBTrait;
+use WP_Rocket\Tests\Integration\TestCase;
+
+/**
+ * @covers \WP_Rocket\Engine\Optimization\RUCSS\Admin\Subscriber::cron_clean_rows
+ *
+ * @group  RUCSS
+ */
+class Test_CronCleanRows extends TestCase{
+	use DBTrait;
+
+	private $input;
+
+	public static function setUpBeforeClass(): void {
+		self::installFresh();
+
+		parent::setUpBeforeClass();
+	}
+
+	public static function tearDownAfterClass() {
+		parent::tearDownAfterClass();
+
+		self::uninstallAll();
+	}
+
+	public function tearDown() : void {
+		remove_filter( 'pre_get_rocket_option_remove_unused_css', [ $this, 'set_rucss_option' ] );
+
+		parent::tearDown();
+	}
+
+	/**
+	 * @dataProvider configTestData
+	 */
+	public function testShouldDoExpected( $input ){
+		$container              = apply_filters( 'rocket_container', null );
+		$rucss_usedcss_query   = $container->get( 'rucss_used_css_query' );
+		$rucss_resources_query = $container->get( 'rucss_resources_query' );
+		$current_date          = current_time( 'mysql', true );
+		$old_date              = strtotime( $current_date. ' - 1 month' );
+
+		$this->input = $input;
+		add_filter( 'pre_get_rocket_option_remove_unused_css', [ $this, 'set_rucss_option' ] );
+
+		$count_remain_used_css = 0;
+		foreach ( $input['used_css'] as $used_css ) {
+			if ( $old_date <  strtotime( $used_css['last_accessed']) ) {
+				$count_remain_used_css ++;
+			}
+			$rucss_usedcss_query->add_item( $used_css );
+		}
+
+		$result_used_css = $rucss_usedcss_query->query();
+		$this->assertCount( count( $input['used_css'] ), $result_used_css );
+
+
+		$count_remain_resources = 0;
+		foreach ( $input['resources'] as $resource ) {
+			if ( $old_date <  strtotime( $resource['last_accessed']) ) {
+				$count_remain_resources ++;
+			}
+			$rucss_resources_query->add_item( $resource );
+		}
+
+		$result_resources = $rucss_resources_query->query();
+		$this->assertCount( count( $input['resources'] ), $result_resources );
+
+		do_action( 'rocket_rucss_clean_rows_time_event' );
+
+		$rucss_usedcss_query       = $container->get( 'rucss_used_css_query' );
+		$rucss_resources_query     = $container->get( 'rucss_resources_query' );
```

Comment:
Why defining them again here?
they are already defined at the start of this test method.

Reply:
@engahmeds3ed we need to get both again from the container because BerlinDB has kinda caching system in place.
So when deleting, the cache value is not deleted so I need to grab a new instance of the Query to retrieve the current values.

Reply:
Due to the previous reason. BerlinDB has caching system. So to re-read the values from the DB I need to grab a new instance of the Query.